unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -741,6 +742,21 @@ public static UTF8String concatWs(UTF8String separator, UTF8String... inputs) {
     return res;
   }
 
+  // TODO: Need to use `Code Point` here instead of Char in case the character longer than 2 bytes
```

Comment:
Not sure if I can get you. 
For the byte version you mean passing in a Map of Array[Byte] -> Array[Byte]? 
4 byte for int, would it be acceptable for a code point of UTF8?